gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -52,15 +54,18 @@ public void apply(Project project) {
 
         Provider<Directory> generatedJavaOutputDir = metricSchemaDir.map(file -> file.dir("generated_src"));
         TaskProvider<GenerateMetricSchemaTask> generateMetricsTask = project.getTasks()
-                .register("generateMetrics", GenerateMetricSchemaTask.class, task -> {
+                .register(GENERATE_METRICS, GenerateMetricSchemaTask.class, task -> {
                     task.setGroup(TASK_GROUP);
                     task.setDescription("Generates bindings for producing well defined metrics");
                     task.getInputFile().set(compileSchemaTask.flatMap(CompileMetricSchemaTask::getOutputFile));
                     task.getOutputDir().set(generatedJavaOutputDir);
                 });
-        project.getTasks().named("compileJava", compileJava -> compileJava.dependsOn(generateMetricsTask));
 
-        configureJavaSource(project, generatedJavaOutputDir);
+        SourceSet mainSourceSet = getMainSourceSet(project);
+        mainSourceSet.getJava().srcDirs(generateMetricsTask);
+        project.getTasks()
+                .named(mainSourceSet.getCompileJavaTaskName())
+                .configure(compileJava -> compileJava.dependsOn(generateMetricsTask));
```

Comment:
Is this explicit `dependsOn` necessary when we declare `mainSourceSet.getJava().srcDirs(generateMetricsTask);` above?